### PR TITLE
Renames `frame` crate to `polkadot-sdk-frame`

### PR DESCRIPTION
**Pull request by gupnik**
Step in https://github.com/paritytech/polkadot-sdk/issues/3155

Needed for https://github.com/paritytech/eng-automation/issues/6

This PR renames `frame` crate to `polkadot-sdk-frame` as `frame` is not available on crates.io

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5633849

File: docs/sdk/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ workspace = true
 # Needed for all FRAME-based code
 parity-scale-codec = { version = "3.0.0", default-features = false }
 scale-info = { version = "2.6.0", default-features = false }
-frame = { path = "../../substrate/frame", features = [
+polkadot-sdk-frame = { path = "../../substrate/frame", features = [
```
Comment (ggwpez) [inline docs/sdk/Cargo.toml:20]:
```suggestion
frame = { package = "polkadot-sdk-frame", path = "../../substrate/frame", features = [
```
Then you dont have to change the code.

Comment (gupnik) [reply docs/sdk/Cargo.toml:20]: Yeah, @rzadp tried it in the linked issue https://github.com/paritytech/eng-automation/issues/6#issuecomment-2012428997 but it didn’t work for the released crate.

**Comment by xlc** [comment]
if we are going to do a rename, can we rename everything? e.g. all the `staging-` creates

**Comment by gupnik** [comment]
> if we are going to do a rename, can we rename everything? e.g. all the `staging-` creates

I will keep this one confined to the frame crate but please open an issue for the others that you would like to be renamed.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5652336

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5652484

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5652632

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5652782

**Comment by liamaharon** [comment]
> if we are going to do a rename, can we rename everything? e.g. all the `staging-` creates

I think for the ones with the `staging-` prefix we are actually in the process of recovering the rights to the original crate name. I don't think we should add to confusion by creating a new intermediate one.

**Review by kianenigma — COMMENTED** [review]
I would say just import the crate with `package = "poladkto-sdk-frame"` and then the code more or less remain the same everywhere.

wdyt @ggwpez @liamaharon?  

Similarly, in all template codes, we should still retain `frame::prelude::*` etc.

**Comment by gupnik** [comment]
> I would say just import the crate with `package = "poladkto-sdk-frame"` and then the code more or less remain the same everywhere.
> 
> wdyt @ggwpez @liamaharon?
> 
> Similarly, in all template codes, we should still retain `frame::prelude::*` etc.

I have added the support for importing both `polkadot-sdk-frame` and its renamed version in the macros. Let me know if this works.

File: prdoc/pr_3813.prdoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Renames `frame` crate to `polkadot-sdk-frame`
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR renames `frame` crate to `polkadot-sdk-frame` as `frame` is not available on crates.io
```
Comment (kianenigma) [inline prdoc/pr_3813.prdoc:9]: Please note that the crate can only be imported as `polkadot-sdk-frame` or `frame`, either here or even better in the crate docs for `frame`.

File: substrate/frame/support/procedural/tools/src/lib.rs [hunk 3]
```diff
@@ -54,15 +54,18 @@ pub fn generate_crate_access(unique_id: &str, def_crate: &str) -> TokenStream {
 ///
 /// This will usually check the output of [`generate_access_from_frame_or_crate`].
 /// We want to know if whatever the `path` takes us to, is exported from `frame` or not. In that
-/// case `path` would start with `frame`, something like `frame::x::y:z`.
+/// case `path` would start with `frame`, something like `polkadot_sdk_frame::x::y:z`.
```
Comment (kianenigma) [inline substrate/frame/support/procedural/tools/src/lib.rs:57]:
```suggestion
/// case `path` would start with `frame`, something like `polkadot_sdk_frame::x::y:z` or frame::x::y:z.
```

File: substrate/frame/support/procedural/tools/src/lib.rs [hunk 4]
```diff
@@ -114,7 +117,9 @@ pub fn generate_hidden_includes(unique_id: &str, def_crate: &str) -> TokenStream
 /// Generates the path to the frame crate deps.
 fn get_frame_crate_path(def_crate: &str) -> Option<syn::Path> {
 	// This does not work if the frame crate is renamed.
-	if let Ok(FoundCrate::Name(name)) = crate_name(&"frame") {
+	if let Ok(FoundCrate::Name(name)) =
+		crate_name(&"polkadot-sdk-frame").map_err(|_| crate_name(&"frame"))
```
Comment (kianenigma) [inline substrate/frame/support/procedural/tools/src/lib.rs:121]:
```suggestion
		crate_name(&"polkadot-sdk-frame").or_else(|_| crate_name(&"frame"))
```

not sure about the return type of `crate_name` but using `or_else` seems correct to me here.

File: substrate/primitives/api/proc-macro/src/utils.rs [hunk 5]
```diff
@@ -34,7 +34,9 @@ pub fn generate_crate_access() -> TokenStream {
 			quote!(#renamed_name::__private)
 		},
 		Err(e) =>
-			if let Ok(FoundCrate::Name(name)) = crate_name(&"frame") {
+			if let Ok(FoundCrate::Name(name)) =
+				crate_name(&"polkadot-sdk-frame").map_err(|_| crate_name(&"frame"))
```
Comment (kianenigma) [inline substrate/primitives/api/proc-macro/src/utils.rs:38]: Same

File: substrate/utils/frame/rpc/support/src/lib.rs [hunk 6]
```diff
@@ -161,7 +161,7 @@ impl<V: FullCodec> StorageQuery<V> {
 
 	/// Send this query over RPC, await the typed result.
 	///
-	/// Hash should be `<YourRuntime as frame::Config>::Hash`.
+	/// Hash should be `<YourRuntime as polkadot_sdk_frame::Config>::Hash`.
```
Comment (kianenigma) [inline substrate/utils/frame/rpc/support/src/lib.rs:164]:
```suggestion
	/// Hash should be `<YourRuntime as frame_system::Config>::Hash`.
```

**Review by kianenigma — APPROVED** [review]
I agree with the direction of allowing `polkadot-sdk-frame`, and letting it be re-imported as `frame`. need better tests to ensure exactly these two scenarios are supported

File: substrate/frame/support/procedural/src/pallet/parse/config.rs [hunk 7]
```diff
@@ -517,52 +518,60 @@ mod tests {
 	#[test]
 	fn has_expected_system_config_works_with_frame() {
 		let frame_system =
-			syn::parse2::<syn::Path>(quote::quote!(frame::deps::frame_system)).unwrap();
```
[the remark this answers is not in the document]
Comment (gupnik) [reply substrate/frame/support/procedural/src/pallet/parse/config.rs:520]: Yeah, that would work. Updated the tests.

**Review by ggwpez — APPROVED** [review]

**Pull request merged**